atomics/T1016/T1016.yaml

```diff
@@ -122,4 +122,13 @@ atomic_tests:
     cleanup_command: |
       Remove-Item -ErrorAction ignore "#{output_file}"
     name: powershell
-
+- name: List macOS Firewall Rules
+  description: "This will test if the macOS firewall is enabled and see what rules are configured. Must be run with elevated privileges.\n\nUpon successful execution, the command will output global firewall configuration and rules per service and port. Additional arguments can be added to see filtered details, such as `globalstate` for global configuration (\"Is it on or off?\"), `firewall` for common application allow rules, and `explicitauths` for specific rules configured by the user. "
+  supported_platforms:
+  - macos
+  executor:
+    command: |
+      sudo defaults read /Library/Preferences/com.apple.alf
```

Comment:
I'm more keen on using this.
`sudo /usr/libexec/ApplicationFirewall/socketfilterfw --getglobalstate`
and 
`sudo /usr/libexec/ApplicationFirewall/socketfilterfw --listapps`

It's possible that more rules exist than what are in the `defaults`

Reply:
I'm fine with either one, I'll keep it consistent with whatever your preferences are but there might be value in both? This will be coupled with some work for sigma rules to detect this activity and it makes sense to have sigma rules for both I'd think, since they both seem to display the most useful information (global firewall state, trusted apps, explicit auths) and I've never run across anything I've seen in the wild using socketfilterfw, but I see defaults used for reads and writes a bit.

Reply:
I agree I think both would be valuable. Ill go ahead and approve my part of this one. Ill let @mattifestation finish the windows stuff and he can merge it.

Reply:
I have added socketfilterfw and cleaned up the description to be more consistent with the formatting i'm seeing else where in this file in https://github.com/redcanaryco/atomic-red-team/pull/1240/commits/fff4ea260b90fa620bb9635a399314c20f470ffc. Thanks for your time reviewing this.